nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelReader.java

```diff
@@ -0,0 +1,193 @@
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private AtomicReferenceArray<String> desiredSheets;
+    private volatile int firstRow;
+    private volatile String dateFormat;
+    private volatile String timeFormat;
+    private volatile String timestampFormat;
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.firstRow = getFirstRow(context);
+        String[] rawDesiredSheets = getRawDesiredSheets(context);
+        this.desiredSheets = new AtomicReferenceArray<>(rawDesiredSheets.length);
+        IntStream.range(0, rawDesiredSheets.length)
+                .forEach(index -> this.desiredSheets.set(index, rawDesiredSheets[index]));
+        this.dateFormat = context.getProperty(DateTimeUtils.DATE_FORMAT).getValue();
+        this.timeFormat = context.getProperty(DateTimeUtils.TIME_FORMAT).getValue();
+        this.timestampFormat = context.getProperty(DateTimeUtils.TIMESTAMP_FORMAT).getValue();
+    }
+
+    private int getFirstRow(final PropertyContext context) {
+        int rawFirstRow = context.getProperty(FIRST_ROW_NUM).asInteger();
+        return getZeroBasedIndex(rawFirstRow);
+    }
+
+    static int getZeroBasedIndex(int rawFirstRow) {
+        return rawFirstRow > 0 ? rawFirstRow - 1 : 0;
+    }
+    private String[] getRawDesiredSheets(final PropertyContext context) {
+        final String desiredSheetsDelimited = context.getProperty(DESIRED_SHEETS).getValue();
+        return getRawDesiredSheets(desiredSheetsDelimited, getLogger());
+    }
+
+    static String[] getRawDesiredSheets(String desiredSheetsDelimited, ComponentLog logger) {
+        if (desiredSheetsDelimited != null) {
+            String[] delimitedSheets = StringUtils.split(desiredSheetsDelimited, ",");
+            if (delimitedSheets != null) {
+                return delimitedSheets;
+            } else {
+                if (logger != null) {
+                    logger.debug("Excel document was parsed but no sheets with the specified desired names were found.");
+                }
+            }
+        }
+
+        return new String[0];
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
+        properties.add(DESIRED_SHEETS);
+        properties.add(FIRST_ROW_NUM);
+        properties.add(DateTimeUtils.DATE_FORMAT);
+        properties.add(DateTimeUtils.TIME_FORMAT);
+        properties.add(DateTimeUtils.TIMESTAMP_FORMAT);
+
+        return properties;
+    }
+
+    @Override
+    protected SchemaAccessStrategy getSchemaAccessStrategy(final String allowableValue, final SchemaRegistry schemaRegistry, final PropertyContext context) {
+        if (allowableValue.equalsIgnoreCase(HEADER_DERIVED.getValue())) {
```

Comment:
It is better to use `equals()` for property value comparisons, since the case should always match.
```suggestion
        if (HEADER_DERIVED.getValue().equals(allowableValue)) {
```